nchelpers/__init__.py

```diff
@@ -293,27 +297,60 @@ def dim_axes(self, dim_names=None):
             return {}
 
         # Start with our best guess
-        axis_to_dim = self.dim_axes_from_names(dim_names)
+        dim_name_to_axis = self.dim_axes_from_names(dim_names)
 
         # Then fill in the rest from the 'axis' attributes
-        # TODO: Does this happen? i.e., when are dimension names the same as axis names?
-        # Alternatively, is this some kind of (relatively benign) programming error?
-        for axis in axis_to_dim.keys():
-            if axis in self.dimensions and axis in self.variables \
-                    and hasattr(self.variables[axis], 'axis'):
-                axis_to_dim[axis] = self.variables[axis].axis
+        for dim_name in dim_name_to_axis.keys():
+            if dim_name in self.dimensions and dim_name in self.variables \
+                    and hasattr(self.variables[dim_name], 'axis'):
```

Comment:
I'm a bit concerned about conflating the conditionals of "variable exists" and "axis attribute exists". I.e. if the variable exists, but doesn't have an axis attribute, then this script will just skip it. However, I would want the script to either error out or raise a really loud warning. We should probably write an issue for this and defer.